### ISSUE TITLE
Add Windows CI via AppVeyor

### ISSUE THREAD

**Issue by sschuberth** [libgit2/rugged#685]
The Windows build has been [broken](https://github.com/libgit2/rugged/issues/612) multiple [times](https://github.com/libgit2/rugged/issues/669) already, so probably adding CI on Windows via [AppVeyor](https://www.appveyor.com/) in addition to Travis CI is a good idea.

**Comment by sschuberth** [comment]
As [AppVeyor already has MSYS2 installed](https://www.appveyor.com/docs/installed-software/#mingw-msys-cygwin), it probably makes sense looking into using https://github.com/larskanis/rubyinstaller2 instead of https://github.com/oneclick/rubyinstaller.

**Comment by ioquatix** [comment]
I think travis now also has windows support. Not sure how it works though.

**Comment by sschuberth** [comment]
Right, but a possible advantage of using AppVeyor in addition to Travis is that more builds could run in parallel, on different services.

**Comment by ioquatix** [comment]
For open source projects everything runs in parallel AFAIK.

**Comment by ioquatix** [comment]
https://blog.travis-ci.com/2018-10-11-windows-early-release

**Comment by ioquatix** [comment]
Maybe not ready for prime time yet?

**Comment by sschuberth** [comment]
> For open source projects everything runs in parallel AFAIK.

Not everything, but [up to 5 builds](https://stackoverflow.com/a/34809448/1127485) (that information is actually hard to find in the current documentation). Still, that's probably enough for this project, I agree.

**Comment by ethomson** [comment]
We're using [Azure Pipelines](https://azure.com/pipelines) on libgit2 itself, and I'm moving LibGit2Sharp over to that as well.  Moving to a single, container-oriented build that runs on actual hypervisors has been a big win.  I conceded that I'm biased here.  But when we did move, we also stopped paying for an AppVeyor subscription, so I think that you'll see the already rather lengthy queue times increase, not decrease.

**Comment by ethomson** [comment]
I can dig some up, maybe, but in moving, we changed the way we build and test since we have more capability.  Since queue time is dramatically lessened, we have a greater opportunity to do more builds and more tests while still getting results for PR validation builds faster.  The biggest win (for libgit2) is the queue times, having 10 builds start immediately is a lot bigger impact (for us) than a 5% difference in the time it takes to compile (in either direction).
